text/0224-experimental-semantic-conventions.md

```diff
@@ -0,0 +1,115 @@
+Before creating a new attribute, the attribute author SHOULD review the semantic conventions and provisional registry to see if an attribute already exists which fits the intended use case.
+If such a registered attribute exists, then it SHOULD be preferred over the creation of new attributes.
+
+An entry MUST NOT be refused without credible reason to believe that registration would be harmful.
+Some reasonable grounds for refusal might be frivolous use of the registry, TC consensus that the attribute may be a security or privacy risk, conflict with another attribute or a name reserved by the TC for future use, or that the proposal is sufficiently lacking in purpose, or misleading about its purpose, that it can be held to be a waste of time and effort.
+Note that a disagreement about technical details are not considered grounds to refuse listing in the provisional registry.
+
+A provisional registry entry which has gained enough support, use, and market penetration SHOULD be promoted to the semantic conventions following existing procedures for review and inclusion.
+When proposing a new semantic convention, existing provisional registry entries SHOULD be considered, and a semantic convention proposal SHOULD NOT break compatibility with an existing provisional registry entry without sufficient reasoning.
+If a semantic convention is added which is not compatible with existing provisional registry entries, a new name SHOULD be created.
+For example, if a provisional entry exists for http.headers, a semantic convention proposal which breaks compatibility might choose the name http.message_headers in order to avoid a conflict.
+
+This document explicitly does NOT define a process for removal of a registered attribute from the provisional registry or semantic conventions.
+Because absence of evidence does not constitute evidence of absence, there is no way to know if any registered attribute is in active use.
+If a registered attribute is deemed by the TC to be harmful in some way, it SHOULD be marked deprecated, and sufficient reasoning for the deprecation should be included in the registry.
+
+Consumers of telemetry SHOULD consider attributes and their values to be untrusted input.
+Attribute values SHOULD be validated before assuming they conform to the format specified in the semantic conventions or the provisional registry.
+Consumers SHOULD consider any security and privacy implications of displaying unhidden attribute values.
+
+## Alternatives Considered
+
+### "x." prefix
+
+This alternative involves establishing an “x.” prefix convention for experimental conventions.
+For example, "x.mydatabase.attribute" may be an experimental version of the eventual "mydatabase.attribute" attribute key.
+See [X- convention for HTTP message headers](#x--http-header-convention) below for more details about why this alternative was discarded.
+
```

Comment:
I'd suggest the following approach as an  alternative:
1. having an otel-specification-contrib repo where semantic conventions can mature and evolve as a whole with less scrutiny or stability expectations. 
  - Otel-owned instrumentation might exist for them (in corresponding -contrib) repos, but would be in a preview state
3. tech-specific specs can live there as well
4. mature general-purpose semantic conventions can be sent to -spec repo

Reply:
> I'd suggest the following approach as an alternative:
> 
> 1. having an otel-specification-contrib repo where semantic conventions can mature and evolve as a whole with less scrutiny or stability expectations.
>    
>    * Otel-owned instrumentations might exist for them (in the corresponding -contrib) repos

I'm not really trying to encourage instability of attributes as part of the experimentation process. Instead of a breaking change for a particular attribute I would rather see a new attribute registered with a different name if the need arises.

> 2. semantic conventions are versioned individually with semver and it reflects their status (stable, beta, alpha, etc). version is populated on the telemetry as part of schema URL that makes status explicit
>    
>    * otel-owned instrumentations  share the same status (stable, beta, alpha) as the conventions they implement

I think there are 2 things here which need to be versioned: the individual attributes and the semantic conventions which use them.

> 3. tech-specific specs can live there as well
> 4. mature general-purpose semantic conventions can be sent to -spec repo

I see value in your proposal here, but I think it covers a slightly different target than I was going for. I could see the two solutions both being implemented eventually.

Reply:
I think 99% of the value would be covered by simply registering attribute names (agnostic of signal type). As JS maintainer most of the instrumentation requests I get are "please add `x` data to the span," or "please add a hook/configuration to `y` instrumentation so I can add arbitrary attributes to the span." Usually if the requested data is not covered by semconv we don't know what to do. There are already many instances "in the wild" of users adding custom attributes using whatever key made sense to them because it wasn't covered by semconv. If there was a registry of these keys at least users who add custom attributes might do so in a consistent manner.

In terms of span kind and parent/child/link structure, we really don't see many requests (at least in JS)

Reply:
I believe the newly limited scope of the proposal to an attribute dictionary should address this concern. Please resolve this conversation if you agree.